src/DataTypes/Serializations/ISerialization.h

```diff
@@ -25,12 +33,30 @@ class Field;
 struct FormatSettings;
 struct NameAndTypePair;
 
+/** Represents serialization of data type.
+ *  Has methods to serialize/deserialize column in binary and several text formats.
+ *  Every data type has default serialization, but can be serialized in different representations.
+ *  Default serialization can be wrapped to one of the special kind of serializations.
+ *  Currently there is only one special serialization: Sparse.
```

Comment:
I am somewhat confused by this: so Sparse is both an in-memory column format, and a serialization format? We probably have to state this explicitly and describe which permutations are possible, i.e. can we serialize a vector column as sparse, or serialize a sparse column as normal one.

Reply:
Yes, Sparse is both in-memory column format, and a serialization format. Initially all columns are full and we serialize full column in sparse serialization, but deserialization is possible only to `ColumnSparse`.